cpp/tests/copying/get_value_tests.cpp

```diff
@@ -166,5 +166,285 @@ TYPED_TEST(DictionaryGetValueTest, GetNull)
+  strings_column_wrapper expected_data{};
+  size_type index = 1;
+
+  auto s       = get_element(col, index);
+  auto typed_s = static_cast<list_scalar const*>(s.get());
+
+  EXPECT_TRUE(s->is_valid());
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_data, typed_s->view());
+}
+
+TEST_F(ListGetStringValueTest, NonNestedGetNull)
+{
+  using LCW = cudf::test::lists_column_wrapper<string_view>;
+
+  std::vector<valid_type> valid{1, 0, 0, 1};
+  LCW col({LCW{"aaa", "Héllo"}, LCW{}, LCW{""}, LCW{"42"}}, valid.begin());
+  size_type index = 2;
+
+  auto s = get_element(col, index);
+
+  EXPECT_FALSE(s->is_valid());
+}
+
+TEST_F(ListGetStringValueTest, NestedGetNonNullNonEmpty)
+{
+  using LCW = cudf::test::lists_column_wrapper<string_view>;
+
+  // clang-format off
+  LCW col{
+    LCW{LCW{"aaa", "Héllo"}},
+    LCW{},
+    LCW{LCW{""}},
+    LCW{LCW{"42"}, LCW{"21"}}
+  };
+  // clang-format on
+  LCW expected_data{LCW{""}};
+  size_type index = 2;
+
+  auto s       = get_element(col, index);
+  auto typed_s = static_cast<list_scalar const*>(s.get());
+
+  EXPECT_TRUE(s->is_valid());
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_data, typed_s->view());
+}
+
+TEST_F(ListGetStringValueTest, NestedGetNonNullNonEmptyPreserveNull)
+{
+  using LCW = cudf::test::lists_column_wrapper<string_view>;
+
+  std::vector<valid_type> valid{0, 1};
+  // clang-format off
+  LCW col{
+    LCW{LCW{"aaa", "Héllo"}},
+    LCW{},
+    LCW({LCW{""}, LCW{"cc"}}, valid.begin()),
+    LCW{LCW{"42"}, LCW{"21"}}
+  };
+  // clang-format on
+  LCW expected_data({LCW{""}, LCW{"cc"}}, valid.begin());
+  size_type index = 2;
+
+  auto s       = get_element(col, index);
+  auto typed_s = static_cast<list_scalar const*>(s.get());
+
+  EXPECT_TRUE(s->is_valid());
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected_data, typed_s->view());
+}
+
+TEST_F(ListGetStringValueTest, NestedGetNonNullEmpty)
+{
+  using LCW = cudf::test::lists_column_wrapper<string_view>;
+
+  // clang-format off
+  LCW col{
+    LCW{LCW{"aaa", "Héllo"}},
+    LCW{LCW{""}},
+    LCW{LCW{"42"}, LCW{"21"}},
+    LCW{}
+  };
+  // clang-format on
+  LCW expected_data{};
+  size_type index = 3;
+
+  auto s       = get_element(col, index);
+  auto typed_s = static_cast<list_scalar const*>(s.get());
+
+  EXPECT_TRUE(s->is_valid());
+  // Relax to equivalent. `expected_data` leaf string column does not
+  // allocate offset and byte array, but `typed_s` does.
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expected_data, typed_s->view());
+}
+
+TEST_F(ListGetStringValueTest, NestedGetNull)
+{
+  using LCW = cudf::test::lists_column_wrapper<string_view>;
+
+  std::vector<valid_type> valid{0, 0, 1, 1};
+  // clang-format off
+  LCW col(
+    {
+      LCW{LCW{"aaa", "Héllo"}},
+      LCW{LCW{""}},
+      LCW{LCW{"42"}, LCW{"21"}},
+      LCW{}
+    }, valid.begin());
+  // clang-format on
+  LCW expected_data{};
+  size_type index = 0;
+
+  auto s = get_element(col, index);
+  EXPECT_FALSE(s->is_valid());
+}
+
```

Comment:
Some additional tests needed:

- More cases with validity, particularly validity on the inner lists (null int, strings, etc).
- Structs, including list<struct<...>>,   list<struct<list<> ...>>  (this will require the gather change suggested above.

Reply:
`list<structs<list<>>>` tests added. `copy_slice` turns out to work just fine with `struct` columns. Also sprinkled some null values into existing fixed-width/string tests.